crates/forge/tests/e2e/common/runner.rs

```diff
@@ -205,3 +207,85 @@ pub(crate) fn get_current_branch() -> String {
         String::from_utf8(output.stdout).unwrap().trim().to_string()
     }
 }
+
+pub(crate) fn intersection_with_wildcard(
+    asserted: &Vec<String>,
+    actual: &Vec<String>,
+) -> Vec<(String, usize)> {
+    let mut result = vec![];
+
+    for element in asserted {
+        let escaped = regex::escape(element);
+        let replaced = escaped.replace("\\[\\.\\.\\]", ".*");
+        let wrapped = format!("^{replaced}$");
+        let re = Regex::new(wrapped.as_str()).unwrap();
+
+        let index = actual.iter().position(|other| re.is_match(other));
+        if let Some(index) = index {
+            result.push((element.clone(), index));
+        }
+    }
+
+    result
+}
+
+pub(crate) fn is_present(line: &str, actual: &mut Vec<String>, asserted: &Vec<String>) -> bool {
+    let present = intersection_with_wildcard(asserted, actual);
```

Comment:
Why do we need to search for the whole pattern instead of just `line`? I think it would simplify the flow by a lot